olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,229 @@
+                description="Random seed for sampling calibration dataset. Default value is 0.",
+            ),
+            "damp_percent": PassConfigParam(
+                type_=float,
+                default_value=0.01,
+                description="Damping factor for quantization. Default value is 0.01.",
+            ),
+            "static_groups": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use static groups for quantization. Default value is False.",
+            ),
+            "true_sequential": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use true sequential for quantization. Default value is False.",
+            ),
+            "desc_act": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use descriptive activation for quantization. Default value is False.",
+            ),
+            "sym": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Symmetric quantization. Default value is False.",
+            ),
+            "data_config": PassConfigParam(
+                type_=Union[DataConfig, Dict],
+                default_value=None,
+                description="""
+                    Data config for quantization. Default value is None.
+                """,
+            ),
+            # TODO(trajep): consider to use data_config to implement the functionality of dataloader_func.
+            "dataloader_func": PassConfigParam(
+                type_=Union[Callable, str],
+                default_value=None,
+                description="""Function/function name to generate dataset for quantization.
+                The returned datasets is a list of tokenized data
+                (e.g. [{ 'input_ids': [ 1, 100, 15, ... ],'attention_mask': [ 1, 1, 1, ... ]},...]).
+                Default is None.
+                """,
+            ),
+            "dataloader_func_kwargs": PassConfigParam(
+                type_=Dict[str, Any],
+                default_value=None,
+                description="Keyword arguments for dataloader_func. Default value is None.",
+            ),
+            "use_dynamo_exporter": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Whether to use dynamo_export API to export ONNX model. Default value is False.",
+            ),
+            "target_opset": PassConfigParam(
+                type_=int,
+                default_value=13,
+                description="The version of the default (ai.onnx) opset to target. Default value is 13.",
+            ),
+            "export_device": PassConfigParam(
+                type_=str,
+                default_value="cpu",
+                description="The device to export the model. Default value is 'cpu'.",
+            ),
```

Comment:
@jambayk , since the pass always be run in CUDA, do we still need this one?

Reply:
I guess jambay is worried about memory unsufficient when export. During quantization, model is loaded into gpu layer by layer, but when export it may consume more memory than that? And it may cause OOM?

Reply:
it seems the conversion only transfer the model and dummy inputs to CUDA in this case. Would this consume extra CUDA memories?

Reply:
Yes, quantization only put one layer into cuda, but conversion may put whole model into cuda, so conversion may consume more memories than quantization, and user may succeed in quantization but fail in conversion.
auto_gptq:
https://github.com/AutoGPTQ/AutoGPTQ/blob/896d8204bc89a7cfbda42bf3314e13cf4ce20b02/auto_gptq/modeling/_base.py#L409-L415
conversion:
https://github.com/Taka152/Olive/blob/2361389e82bd8e468ce981603c26612c352bb267/olive/passes/onnx/conversion.py#L176

Reply:
yes, like @Taka152 said conversion on GPU takes more memory since it's not just putting input and model on GPU but also needs to do a forward pass. The forward pass, depending on the implementation can consume much more memory.

Reply:
could we try GPU firstly. If out of memory then fallback to CPU? or alternatively, just do conversion on CPU? in this case, avoid specify the convert device since user really doesn't care it?

Reply:
I think default arguments won't bring comfusion to user, and useful for advanced users, like extra_args in optimum_conversion.

OOM catching seems cool, and we could try it in another pr? Since other conversions can also benefit from it. Like user specify cuda but encounter OOM, we can fallback to cpu to do export. 

I'm ok with just do conversion on cpu since it seems most safest method.

Reply:
BTW, it seems the config name is different with that one used in conversion? https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/conversion.py#L75

Reply:
Yes, I changed that since user may think the use_device is used for quantization rather than export. But in conversion pass use_device is clear for export.

Reply:
Inc(device): https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/inc_quantization.py#L32
optimum(device): https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/optimum_conversion.py#L43
sparsegpt(compute_device): https://github.com/microsoft/Olive/blob/main/olive/passes/pytorch/sparsegpt.py#L75
snpe(target_device): https://github.com/microsoft/Olive/blob/main/olive/passes/snpe/snpe_to_onnx.py#L34
conversion(use_device).

@jambayk, @trajepl 
it seems the device names are not consistent. need we use the same name?

Reply:
please refer to https://aiinfra.visualstudio.com/Model%20optimization%20Toolkit/_sprints/taskboard/Model%20optimization%20Toolkit%20Team/Model%20optimization%20Toolkit/[Ge]%20Germanium